tests/scripts/check_names.py

```diff
@@ -0,0 +1,830 @@
+        gutter_len = max(4, len(ln_str))
+        gutter = (gutter_len - len(ln_str)) * " " + ln_str
+        underline = self.pos[1] * " " + (self.pos[2] - self.pos[1]) * "^"
+
+        return (
+            " {0} |\n".format(gutter_len * " ") +
+            " {0} | {1}".format(gutter, self.line) +
+            " {0} | {1}\n".format(gutter_len * " ", underline)
+        )
+
+class Problem(): # pylint: disable=too-few-public-methods
+    """
+    A parent class representing a form of static analysis error.
+    """
+    def __init__(self):
+        self.quiet = False
+        self.textwrapper = textwrap.TextWrapper()
+        self.textwrapper.width = 80
+        self.textwrapper.initial_indent = "    > "
+        self.textwrapper.subsequent_indent = "      "
+
+class SymbolNotInHeader(Problem): # pylint: disable=too-few-public-methods
+    """
+    A problem that occurs when an exported/available symbol in the object file
+    is not explicitly declared in header files. Created with
+    NameCheck.check_symbols_declared_in_header()
+
+    Fields:
+    * symbol_name: the name of the symbol.
+    """
+    def __init__(self, symbol_name):
+        self.symbol_name = symbol_name
+        Problem.__init__(self)
+
+    def __str__(self):
+        if self.quiet:
+            return "{0}".format(self.symbol_name)
+
+        return self.textwrapper.fill(
+            "'{0}' was found as an available symbol in the output of nm, "
+            "however it was not declared in any header files."
+            .format(self.symbol_name))
+
+class PatternMismatch(Problem): # pylint: disable=too-few-public-methods
+    """
+    A problem that occurs when something doesn't match the expected pattern.
+    Created with NameCheck.check_match_pattern()
+
+    Fields:
+    * pattern: the expected regex pattern
+    * match: the Match object in question
+    """
+    def __init__(self, pattern, match):
+        self.pattern = pattern
+        self.match = match
+        Problem.__init__(self)
+
+    def __str__(self):
+        if self.quiet:
+            return (
+                "{0}:{1}:{2}"
+                .format(self.match.filename, self.match.pos[0], self.match.name)
+            )
+
+        return self.textwrapper.fill(
+            "{0}:{1}: '{2}' does not match the required pattern '{3}'."
+            .format(
+                self.match.filename,
+                self.match.pos[0],
+                self.match.name,
+                self.pattern
+            )
+        ) + "\n" + str(self.match)
+
+class Typo(Problem): # pylint: disable=too-few-public-methods
+    """
+    A problem that occurs when a word using MBED doesn't appear to be defined as
+    constants nor enum values. Created with NameCheck.check_for_typos()
+
+    Fields:
+    * match: the Match object of the MBED name in question.
+    """
+    def __init__(self, match):
+        self.match = match
+        Problem.__init__(self)
+
+    def __str__(self):
+        if self.quiet:
```

Comment:
The structure `if self.quiet: ...` is repeating through all children of the `Problem` class.
Have a look at the proposed code commented on line 825.